### DISCUSSION TITLE
CR-Touch

### DISCUSSION THREAD

**Discussion by Metalizer6** [Jyers/Marlin#1846] [Q&A] [answered]
Hello, does the firmware work with Creality CR-Touch?
Thank you

**Comment by MathiasMC** [comment]
Yes im using CR-Touch

**Reply by Sonicboom247** [reply]
I know the CR touch's wiring is all crazy as in red is negative and black is positive, did you have to switch wires for that as well as Z endstop? i just magic smoked a CR touch on my other printer and am just wondering.

**Reply by K2Kevin** [reply]
You must be thinking about the BLTouch wiring.  CRTouch plugs directly into the port, comes with multiple mounts and wiring is perfect length.

**Reply by Sonicboom247** [reply]
Lol i wish, my BLtouch was straight forward connection, my CR Touch on the other hand is all kinds of crazy, there's posts online about the wiring of the red and black needing to be swapped. Maybe not for the ender but for my Hornet it needed to be swapped

**Reply by Metalizer6** [reply]
My CR touch is working flawlessly aswell. Just plug and flash basically

**Comment by K2Kevin** [comment, marked as answer]
> Hello, does the firmware work with Creality CR-Touch?
> Thank you

YES

**Comment by Metalizer6** [comment]
Thank you for answers
